romantic-revolutionaries/modules/navigation/navcont.py

```diff
@@ -0,0 +1,56 @@
+"""Navigation Controller Module"""
+from enum import Enum
+
+
+class Directions(Enum):
+    NORTH = 0
+    EAST = 1
+    SOUTH = 2
+    WEST = 3
+
+
+class NavControl:
+    """Control navigation requests.
+
+        To subscribe pass your callback function to subscribe()
+        Your callback function should accept a direction and a distance.
+
+        If an invalid direction is passed in a ValueError will be raised.
+
+        go() is the public function called to effect navigation.
+        If go() is called without arguments, it will repeat the last
+        direction and distance. If it is called without a distance it will
+        repeat the last distance. And if it is called without a direction
+        it will repeat the last direction. If you want to call with a distance
+        but no direction, you must do so with named argument for your distance.
+    """
```

Comment:
This is a very nice docstring. If the rest of the project had documentation half as good as this, you would have been miles ahead of most of the other teams.